### PR TITLE
Make Navbar Menu Relatively Positioned Instead of Absolutely Positioned

### PR DESCRIPTION
**Pull request by patrickpatrickpatrick**
## What

Make the dropdown menu in layout_super_navigation_header relatively positioned instead of absolutely positioned.

### More details

-  Restructure the HTML to support this change and remove classes/rules that are no longer used. Remove the margin addition from `layout-super-navigation-header.js`.
    * Move the buttons and links into a container which is absolutely positioned. This button container has a parent container that is the same width of the page content
    * Move the menus outside of the button container into their own containers. 
    * Change the menus to be relatively positioned instead of absolutely positioned
    * Remove the unneeded rules and classes now that the menus are relatively positioned instead of absolutely positioned
- Remove the resize listener 

## Why

In the previous implementation of the navbar, the absolute positioning was required for mobile layout. In this current implementation however we can make the positioning of the navbar relative without affecting the page layout. On live currently when the navbar is opened, a margin of the height of the navbar is added to the bottom of the layout_super_navigation_header to push the content of the page down below the menu. This basically simulates what the styling would be if the menu was relatively positioned anyway. This was a change I would have made with the other navbar changes (https://github.com/alphagov/govuk_publishing_components/pull/3058) but due to time constraints it was decided to split this work out.

Aside from the absolute positioning not being necessary anymore there are issues that these changes fix:

- the padding on either side of the navbar on some tablet viewport sizes being incorrect 
- the padding of the search menu on tablet viewports being incorrect
- when CSS is disabled, there is no longer excess margin between the navbar and the content of the page
- restores the bottom border to the search menu

And also means we no longer need to keep listening out for resize events to make sure the layout of the menu is correct.

### Relevant Trello Cards
- https://trello.com/c/os3bpX19/188-navbar-pushes-down-page-content-if-javascript-is-enabled-and-the-stylesheet-fails-to-load
- https://trello.com/c/RC9CYHtm/1573-navbar-has-some-layout-issues-on-android-devices-s

### Relevant Issues

- https://github.com/alphagov/govuk_publishing_components/issues/3194
- https://github.com/alphagov/govuk_publishing_components/issues/3192
- https://github.com/alphagov/govuk_publishing_components/issues/3087

#### Before

Screenshot 2023-01-30 at 11 35 10

Screenshot 2023-01-30 at 12 02 17

#### After

Screenshot 2023-01-30 at 11 47 27

Screenshot 2023-01-30 at 12 02 22

#### Before

Screenshot 2023-01-30 at 11 32 16

#### After

Screenshot 2023-01-30 at 11 31 53

### CODE REVIEW

**Comment by patrickpatrickpatrick** [comment]
Will squash the HTML commits, they are separated for clarity.

**Review by hannalaakso — COMMENTED** [review]
Looks like a good cleanup and will hopefully make the navbar more maintainable going forward 🙌  (I think we shouldn't squash the commits and messages but keep them as they are as so that the changes in the PR can be reviewed by anyone in the future needing to understand what changed, especially where there were large blocks of code being moved around in a single commit.)

Should we add some tests for the new containers added to the markup?

I'm going to do some more cross browser and assistive tech testing to check all the issues and cards flagged in this PR have been fixed 🕵️

File: app/views/govuk_publishing_components/components/_layout_super_navigation_header.html.erb [hunk 1]
```diff
@@ -141,6 +141,9 @@
             </div>
 
             <div class="gem-c-layout-super-navigation-header__search-item">
+              <h3 class="govuk-visually-hidden">
```
[the remark this answers is not in the document]
Comment (patrickpatrickpatrick) [reply app/views/govuk_publishing_components/components/_layout_super_navigation_header.html.erb]: ah i moved this to the wrong place, it's now where it should be with the form

File: app/assets/stylesheets/govuk_publishing_components/components/_layout-super-navigation-header.scss [hunk 2]
```diff
@@ -89,12 +89,26 @@ $after-button-padding-left: govuk-spacing(4);
   }
 }
 
+.gem-c-layout-super-navigation-header__navbar-container {
```
[the remark this answers is not in the document]
Comment (hannalaakso) [reply app/assets/stylesheets/govuk_publishing_components/components/_layout-super-navigation-header.scss]: Thank you, I think the style has now been removed but I can see the markup for the wrapper still there.

**Comment by hannalaakso** [comment]
Thanks for addressing my comments @patrickpatrickpatrick 👍 

@MartinJJones Would you be okay to check if the bugs you raised on the navbar recently are fixed by this PR?

**Comment by MartinJJones** [comment]
Nice work!, @hannalaakso will take a look at this today and check if the navbar bugs are fixed by this change as well

**Comment by MartinJJones** [comment]
Nice work on this, I tested the changes locally and this PR will fix the related issues 👍 

After checking Percy, I noticed a couple of things that could be worth further investigation.

**Menu and search buttons overlap when the viewport width is <769px**

buttons-overlap

**Change to the flow of menu items**

The [visual change shown in Percy](https://percy.io/55ba5e1a/govuk_publishing_components/builds/25012237/changed/1396631241?browser=firefox&browser_ids=33&subcategories=unreviewed%2Cchanges_requested&viewLayout=overlay&viewMode=new&width=375&widths=375%2C1280) looks good to me, in that all menu items will now appear on the same line.

However, we may run into some accessibility/usability issues for users who Zoom in. The screenshot below shows an example of the menu bar zoomed in to 300% ([based on the govuk accessibility statement](https://www.gov.uk/help/accessibility-statement))

Screenshot 2023-02-10 at 11 59 13

It is also worth noting that there appears to be existing accessibility/usability issues with the current menu bar, which this PR also fixes, although not directly in the scope of the changes required.

| Before | After |
|---|---|
| when zoomed in 300% on the live site, and you click the Menu button, you can no longer click the Menu button again as it is hidden behind "Topics" | With the changes in this PR, the Menu button is always visible: |
| Screenshot 2023-02-10 at 12 05 25 | Screenshot 2023-02-10 at 12 07 08 |

**Review by MartinJJones — COMMENTED** [review]
Recent changes look good 👍 

The only other thing I can see that is worth fixing is the increased overlap between the menu and search buttons as shown in the [Percy visual regression test](https://percy.io/55ba5e1a/govuk_publishing_components/builds/25104467/changed/1401171358?browser=firefox&browser_ids=33&subcategories=unreviewed%2Cchanges_requested&viewLayout=overlay&viewMode=new&width=375&widths=375%2C1280)

We should be able to fix this by changing the line below to `margin: 0`

https://github.com/alphagov/govuk_publishing_components/commit/3dd13dc1fe05820c610fe232e64590cb8bf70b6e#diff-81dddb5f2e54064fddcee20cd639b6813b236c4912e2517f9ee93a62453322e0R416

I've tested this as well using Safari 16 on iOS using Browserstack, and appeared to work OK:

Screenshot 2023-02-14 at 13 42 09

**Comment by MartinJJones** [comment]
@patrickpatrickpatrick @hannalaakso The changes look good to me, I've also approved the 2 visual regression test flagged in Percy.

The issues I reported are fixed, I tested the new menu layout in the latest versions of Chrome, Firefox and Safari, and also tested on IE11 in Browserstack which all worked great 👍 

I've not tested the changes using any AssistiveTechnologies however, so I've held off approving the PR just in case you still need to do more testing.

**Review by hannalaakso — APPROVED** [review]
Looks good to me as well, thanks @patrickpatrickpatrick 👍

**Pull request merged**